_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,70 @@
+
+So how does a code-review ideally look like?
+
+Lets base on the assumption that every reviewer has the same intent - which is to convince the reviewee to follow the suggested changes (to improve the quality of the added code)?
+
+When this is the main intent, different approaches to code reviews can only be different strategies to achieve the same goal.
+
+The following strategies/behavioural advices can help to achieve this goal and at the same time improve the team spirit and culture.
+
+# Consult. Dont preach.
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a consultant. Whatever you say or write: do it with caution. Sentences that start with "I would approach the problem this way..." or "I believe this way would be more efficient" are a good approach if you want to encourage the reviewee to accept your recommendations. Way better than "this solution is inappropriate" or "do it like this". Command-style comments are guaranteed to result in damage. Conflicts, drop of creativity and motivation, bad team spirit. Dont do it.
+By the way: adding "please" to such a sentence doesnt help. You already clearly communicated a difference in status. You already switches your consultant-hat for something that is simply inappropriate in modern software development teams. The result will be a bad taste in the mouth at best.
+By the way: the (probably correct) advice will now not be accepted full-heated. And team morale is damaged.
+
+If the reviewer would have invested just a little more time and watched the way he communicated: the reviewee might have been very thankful and might have accepted all advise with a smile.
+
+Your effort is well invested if you take a break from time to time and make sure your mindset is tuned, your communication style is respectful and reflects that you are a buddy who just wants to help.
+
+
+# Do not enforce consensus.
+
+Engineers often tend to forget that there is more than one valid opinion in a lot of cases. Not everything is determined to false and true.
+So if the reviewer has communicated his opinion appropriate and was not able to convince the reviewee, the code review is the wrong forum to let discussion escalate and enforce consensus. If the code is robust and fulfills the requirements, let it pass the review and try to move the topic out of the situation. You may have tech meetings or other forums in which you can discuss the topic (ideally in general and not by specific example) with a broader audience.
+Letting the topic lie for a week can also help. Often times you realize that consensus isnt important at all. That not only there are more valid opinions than yours. But also that your team maybe can afford to let both opinions stand.
+
+# Do not point at rules but give explanations
```

Comment:
I really liked the first title "Consult. Don't preach!" 
What do you say about making all the headlines on the same style? So this would be
> Give explanations. Don't point at rules!